src/components/TodoList.tsx

```diff
@@ -0,0 +1,200 @@
+      event.preventDefault();
+
+      edit(todo);
+    }
+
+    if (event.key === 'Escape') {
+      setActiveChangeId(0);
+    }
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {filteredTodos?.map(todo => (
+        <div
+          data-cy="Todo"
+          className={`todo ${cn({ completed: todo.completed })}`}
+          key={todo.id}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              onClick={() => updateStatus(todo)}
+              checked={todo.completed}
+            />
+          </label>
+
+          {activeChangeId !== todo.id ? (
+            <span
+              data-cy="TodoTitle"
+              className="todo__title"
+              onDoubleClick={() => {
+                setActiveChangeId(todo.id);
+                setEditedTodo(todo.title);
+              }}
+            >
+              {todo.title}
+            </span>
+          ) : (
+            <form onSubmit={event => event.preventDefault()}>
+              <input
+                data-cy="TodoTitleField"
+                type="text"
+                className="todo__title-field"
+                placeholder={todo.title}
+                value={editedTodo}
+                onChange={event => setEditedTodo(event.target.value)}
+                onBlur={() => edit(todo)}
+                onKeyDown={event => handleEdit(event, todo)}
+                autoFocus
+              />
+            </form>
+          )}
+
+          {!activeChangeId && (
+            <button
+              type="button"
+              className="todo__remove"
+              data-cy="TodoDelete"
+              onClick={() => handleDelete(todo.id)}
+            >
+              ×
+            </button>
+          )}
+
+          <div
+            data-cy="TodoLoader"
+            className={`modal overlay ${cn({ 'is-active': loaderId.includes(todo.id) || deletedTodoId?.includes(todo.id) })}`}
+          >
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      ))}
+
+      {Boolean(tempTodo) && (
+        <div data-cy="Todo" className="todo">
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {defaultTempTodo.title}
+          </span>
+
+          <button type="button" className="todo__remove" data-cy="TodoDelete">
+            ×
+          </button>
+          <div data-cy="TodoLoader" className="modal overlay is-active">
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      )}
```

Comment:
you can try reuse Todo component or create new TempTodo

Reply:
Look

Reply:
I`m not sure that I can reuse it when there are event handlers inside